site/docs/v1/tech/oauth/url-validation.adoc

```diff
@@ -0,0 +1,274 @@
+
+ * The domain must contain at least three segments
+ * The wildcard may only appear in the host portion of the domain (left-most subdomain)
+ * The wildcard can be a full or partial replacement of the host
+
+The following table provides some examples of valid and invalid wildcard patterns.
+
+[cols="45m,10,45"]
+.Domain Wildcards
+|===
+| Example | Valid | Reason
+
+|\https://*.example.com
+|✓
+|&nbsp;
+
+|\https://blah*.example.com
+|✓
+|&nbsp;
+
+|\https://*.com
+|✕
+|The domain only contains two segments
+
+|\https://auth.*.com
+|✕
+|The wildcard does not appear in the host
+
+|\https://pass:[*]midpass:[*].example.com
+|✕
+|The domain contains multiple wildcards
+
+|===
+
+=== Port
+
+The port number can be specified as a wildcard. There is no partial wildcard support for the port number.
+
+[cols="45m,10,45"]
+.Port Wildcards
+|===
+| Example | Valid | Reason
+
+|\https://example.com:*
+|✓
+|&nbsp;
+
+|\https://example.com:4*
+|✕
+|Partial wildcards are not allowed for the port number
+
+|===
+
+=== Path Segments
+
+Wildcards are allowed in path segments with the following restrictions:
+
+ * There can be no more than one wildcard _per path segment_
+ * The wildcard can be a full or partial replacement of the path segment
+
+The following table provides some examples of valid and invalid wildcard patterns in the path.
+
+[cols="45m,10,45"]
+.Path Wildcards
+|===
+| Example | Valid | Reason
+
+|\https://example.com/path/*/resource
+|✓
+|&nbsp;
+
+|\https://example.com/p*/to/resource
+|✓
+|&nbsp;
+
+|\https://example.com/pass:[*]/par*tial/pass:[*]
+|✓
+|&nbsp;
+
+|\https://example.com/path/pass:[*]midpass:[*]/resource
+|✕
+|The path segment contains multiple wildcards
+
+|===
+
+=== Query String Values
+
+Wildcards are allowed in query string values with the following restrictions:
+
+ * Partial wildcards are not allowed for query string values
+
+Wildcards are _not_ allowed in query string names. The following table provides some examples of valid and invalid wildcard patterns in the query string.
+
+[cols="45m,10,45"]
+.Query String Wildcards
+|===
+| Example | Valid | Reason
+
+|\https://example.com?foo=*
+|✓
+|&nbsp;
+
+|\https://example.com?foo=pass:[*]&bar=pass:[*]&baz=blah
+|✓
+|&nbsp;
+
+|\https://example.com?foo=par*tial
+|✕
+|Partial wildcard replacement is not allowed for query string values
+
+|\https://example.com?*=blah
+|✕
+|Wildcards are not allowed in query string names
+
+|===
+
+== Wildcard Replacement Patterns
+
+The position where wildcards are allowed in configured values is just one half of the puzzle. Wildcards in each portion of the URL have different rules for the replacement values. Please note that allowed replacement values may not produce a valid URL. This section provides details on the allowed replacements for wildcards in each portion of the URL. A wildcard in the configured value must be replaced with at least one character in the concrete value.
```

Comment:
What is a concrete value? This is the first I've heard of that term.

Reply:
I thought that was common parlance. A configured authorized URL with a wildcard is a pattern or template. The actual value coming in on `redirect_uri` or the `Origin` header would be the concrete value that is matched against the pattern.

I'm open to suggestions for changing this verbiage.

Reply:
could we say:

> .. in the value.

Or 

> ... in the literal value.

Reply:
Was confusing to me :) 

Maybe: 

`A wildcard in the configured value must be replaced with at least one character in the concrete value.` => 

`Each wildcard in the configured value must match one or more characters. Matches against empty strings will fail.`